docs/resources/application_access_grant.md

```diff
@@ -2,32 +2,125 @@
 
 Application Access Grant resource. Purpose of a grant is to request access to a stream in an environment. Read more: https://docs.axual.io/axual/2022.2/self-service/application-management.html#requesting-stream-access
 
+## Note
+- Application Access Grant cannot be edited
+- Cancel grant by destroying application_access_grant
+- Read more: https://docs.axual.io/axual/2023.1/self-service/topic-authorizations.html#contents
+
+## Usage
+- To request access to a stream, create an application_access_grant
+- To cancel a grant, delete the application_access_grant
+- Application Access Grant that are auto authorized do not need any further action.
+- Application Access Grant that required further action can be handled with application_access_grant_approval and application_access_grant_rejection.
+
 <!-- schema generated by tfplugindocs -->
 ## Schema
 
 ### Required
 
-- `access_type` (String) Application Access Type. Accepted values: Consumer, Producer
+- `access_type` (String) Application Access Type. Accepted values: CONSUMER, PRODUCER
 - `application` (String) Application Unique Identifier
 - `environment` (String) Environment Unique Identifier
 - `stream` (String) Stream Unique Identifier
 
 ### Read-Only
 
-- `application_access` (String) Application Access ID to which this Grant belongs
 - `id` (String) Application Access Grant Unique Identifier
 - `status` (String) Status of Application Access Grant
 
 ## Example Usage
 
 ```terraform
 #Logged in Terraform User(by default kubernetes@axual.com) needs to have application admin rights(for create access request) and stream admin rights(for revoking access request) or be owner of the application and the stream (by being user in the same group as the application's and stream's owner group)
-resource "axual_application_access_grant" "terra_grant_1" {
-  application = axual_application.gitops_test_application_2.id
-  stream = axual_stream.gitops_test_stream2.id
-  environment = "7237a4093d7948228d431a603c31c904"
-  access_type = "Consumer"
-  depends_on = [axual_application_principal.test_application_principal, axual_stream_config.gitops_test_stream_config_2, axual_stream.gitops_test_stream2]
+resource "axual_application_access_grant" "dash_consume_from_logs_in_dev" {
```

Comment:
Let's rename axual_application_access_grant to axual_grant please and not mention application_access anywhere in code nor docs. User is not interested in knowing how it works under the hood. For user there is only grant and grant can be approved/rejected/cancelled/revoked.

Reply:
The naming of grants is rather confusing, that's why I made it explicit. 
We call them topic access requests in the UI.

`axual_grant` does not communicate much.

Reply:
I agree `application_access_grant` is more descriptive.